src/librustc_resolve/build_reduced_graph.rs

```diff
@@ -395,7 +400,7 @@ impl<'a> Resolver<'a> {
                 });
                 self.potentially_unused_imports.push(directive);
                 let imported_binding = self.import(binding, directive);
-                if ptr::eq(self.current_module, self.graph_root) {
+                if ptr::eq(parent, self.graph_root) {
```

Comment:
Random aside, is there any chance we could compare modules by `DefId` or something?
Also, the way they are used (for block scopes as well), they feel a bit more like "(def) scopes" than "modules" (but we also have the other kind of scope, the `Rib`s, which can contain runtime variables).

Reply:
`Module` in resolve is either a `mod`, or `enum`, or `trait`, or an `ast::Block`.
The naming seems ancient, I don't know why it was chosen, perhaps there was only `mod` initially, but then it got extended.

Modules and many other things (including those without IDs) are allocated on the resolver arena, so they use referential equality pretty regularly.
In this case `ptr::eq` is just simpler, I guess, than extracting the optional `DefId`s and comparing them if they exist.

Reply:
Oh, wait, two `Module`s can have the same `DefId`, see https://github.com/rust-lang/rust/pull/57181.
I think that case cannot apply to `self.graph_root` specifically, though.